tests/backuptestTriggers.go

```diff
@@ -334,6 +334,11 @@ func eventCreateBackup(inputsForEventBuilder *PxBackupLongevity) (error, string,
 
 func eventRestore(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
 	defer GinkgoRecover()
+	defer func() {
+		log.InfoD("switching to default context")
+		err := SetClusterContext("")
+		log.FailOnError(err, "failed to SetClusterContext to default cluster")
```

Comment:
How about? To avoid log.FailOnError inside a function.

```go
func eventRestore(inputsForEventBuilder *PxBackupLongevity) (err error, restoreName string, eventData EventData) {
    defer GinkgoRecover()
    defer func() {
        log.InfoD("switching to default context")
        errSetContext := SetClusterContext("")
        if errSetContext != nil {
            if err == nil { // Only assign if no previous error
                err = errSetContext
            }
            log.Error("failed to SetClusterContext to default cluster", errSetContext)
        }
    }()

    ctx, errGetCtx := backup.GetAdminCtxFromSecret()
    if errGetCtx != nil {
        err = errGetCtx
        log.Error("Fetching px-central-admin ctx", err)
        return
    }

    eventData = EventData{}

    restoreName = fmt.Sprintf("%s-%s-%s", RestoreNamePrefix, inputsForEventBuilder.BackupData.BackupName, RandomString(5))
    appContextsExpectedInBackup := FilterAppContextsByNamespace(inputsForEventBuilder.ApplicationData.SchedulerContext, inputsForEventBuilder.BackupData.Namespaces)
    err = CreateRestoreWithValidation(ctx, restoreName, inputsForEventBuilder.BackupData.BackupName, make(map[string]string), make(map[string]string), DestinationClusterName, BackupOrgID, appContextsExpectedInBackup)
    if err != nil {
        restoreName = fmt.Sprintf("Restore failed for %s", restoreName) // Update restoreName to indicate failure
        return
    }

    eventData.RestoreName = restoreName
    return
}
```

Reply:
done , will take refactoring other functions in seprate PR.